### ISSUE TITLE
[Android] Retry logic to network layer

### ISSUE THREAD

**Issue by RingerJK** [mapbox/mapbox-navigation-android#3223]
Catching from slack discussion: 
> so networking retries, that logic is really good to have on the networking layer

**Comment by kmadsen** [comment]
Started looking at this some and it MapboxServices could be a good place

**Comment by RingerJK** [comment]
@kmadsen yeah, it's a great option! BTW if we need that we could make it internally in SDK, and later apply external library retry-feature.

**Comment by RingerJK** [comment]
`mapbox-java` issue https://github.com/mapbox/mapbox-java/issues/1164

**Comment by LukasPaczos** [comment]
`MapboxServices` are rather stateless, it's only an abstraction to make _a_ request. IMO, if we were to implement a retry logic, SDK should be responsible for driving this experience and letting the dev customize it.

**Comment by RingerJK** [comment]
>IMO, if we were to implement a retry logic, SDK should be responsible for driving this experience and letting the dev customize it.

actually I was thinking about it. And retry logic might be handled by SDK, agree. But still would be great to expose `MapboxServices` API to make retry calls.
